doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -80,11 +80,14 @@ described elsewhere
 
 .. cmd:: Check @term
 
-   Displays the type of :n:`@term`. When called in proof mode, the
-   term is checked in the local context of the selected goal.
+   Displays the type of :n:`@term`. When called in proof mode, the term is
+   checked in the local context of the selected goal. This command will try
+   resolve an existential variables.
```

Comment:
```suggestion
   checked in the local context of the selected goal. This command tries
   to resolve existential variables.
```

Is there a short and accurate description of how Evars get resolved by `Check`?  Could you perhaps give a short example?  (so I can understand and perhaps to include in the doc.)

Reply:
```coq
Class SomeType := {x : nat}.
#[export] Instance default_term : SomeType := {x := 0}.
Axiom some_predicate : SomeType -> Prop.
Axiom some_proof : forall (t : SomeType), some_predicate t.

Check (some_proof _).
(* some_proof default_term
     : some_predicate default_term *)

Type (some_proof _).
(* some_proof default_term
     : some_predicate default_term *)
```

It appears that `Type` does in fact resolve evars so my documentation of it is incorrect as `@SkySkimmer` says. The difference is this:

```coq
Class SomeType := {x : nat}.
(* #[export] Instance default_term : SomeType := {x := 0}. *)
Axiom some_predicate : SomeType -> Prop.
Axiom some_proof : forall (t : SomeType), some_predicate t.

Check (some_proof _).
(* some_proof ?t
     : some_predicate ?t
where
?t : [ |- SomeType] *)

Type (some_proof _).
(* Cannot infer this placeholder of type
"SomeType" (no type class instance
found). *)

```

Both resolve evars, however `Check` will not fail when evars are left, whereas `Type` will.